src/deploy-cromwell-on-azure/Deployer.cs

```diff
@@ -800,94 +795,86 @@ private async Task<ManagedCluster> GetExistingAKSClusterAsync(string aksClusterN
             })
             .Where(a => a is not null)
             .SelectMany(a => a)
+            .SelectAwaitWithCancellation(async (a, ct) => (await a.GetAsync(ct)).Value)
             .SingleOrDefaultAsync(a =>
-                    a.Name.Equals(aksClusterName, StringComparison.OrdinalIgnoreCase) &&
-                    a.Location.Equals(configuration.RegionName, StringComparison.OrdinalIgnoreCase),
+                    a.Data.Name.Equals(aksClusterName, StringComparison.OrdinalIgnoreCase) &&
+                    a.Data.Location.Name.Equals(configuration.RegionName, StringComparison.OrdinalIgnoreCase),
                 cts.Token);
         }
 
-        private async Task<ManagedCluster> ProvisionManagedCluster(IResource resourceGroupObject, IIdentity managedIdentity, IGenericResource logAnalyticsWorkspace, INetwork virtualNetwork, string subnetName, bool privateNetworking)
+        private async Task<ContainerServiceManagedClusterResource> ProvisionManagedClusterAsync(IResource resourceGroupObject, IIdentity managedIdentity, IGenericResource logAnalyticsWorkspace, INetwork virtualNetwork, string subnetName, bool privateNetworking, string nodeResourceGroupName)
         {
-            var resourceGroup = resourceGroupObject.Name;
+            var uami = (await armClient.GetUserAssignedIdentityResource(new(managedIdentity.Id)).GetAsync(cts.Token)).Value;
+            var resourceGroup = armClient.GetResourceGroupResource(new(resourceGroupObject.Id));
             var nodePoolName = "nodepool1";
-            var containerServiceClient = new ContainerServiceClient(azureCredentials) { SubscriptionId = configuration.SubscriptionId, BaseUri = new Uri(azureCloudConfig.ResourceManagerUrl) };
-            var cluster = new ManagedCluster
+            ContainerServiceManagedClusterData cluster = new(new(configuration.RegionName))
```

Comment:
do you like this syntax?  It seems less readable.  Now I can't tell what the second `new` type is... what am I new'ing up?

Reply:
It obviously accepts a string region name, so it has to be whatever the location-like object is that is the only parameter (which happens to be required) for the only public accessible constructor for the ContainerServiceManagedClusterData class!